API/RecommendationService/__init__.py

```diff
@@ -90,42 +82,47 @@ def main(req: func.HttpRequest) -> func.HttpResponse:
 
 
 async def get_recommendation_items(command_list, recommend_type, error_info, correlation_id, subscription_id, cli_version, user_id, command_top_num=5, scenario_top_num=5):
-    loop = asyncio.get_event_loop()
+    command_list = load_command_list(command_list)
+    success_command_list = get_success_commands(command_list)
 
     # Take the data of knowledge base first, when the quantity of knowledge base is not enough, then take the data from calculation and Aladdin
-    knowledge_base_items_future = loop.run_in_executor(None, get_recommend_from_knowledge_base, command_list, recommend_type, error_info)
+    knowledge_base_items_task = asyncio.create_task(asyncio.to_thread(get_recommend_from_knowledge_base, command_list, recommend_type, error_info))
 
     # Get the recommendation of offline caculation from offline data
-    async def _get_offline_recommendation(command_list, recommend_type, error_info, command_top_num):
-        offline_items = []
-        if need_offline_recommendation(recommend_type, error_info=None):
-            offline_items = await get_recommend_from_offline_data(command_list, recommend_type, error_info=None, top_num=command_top_num)
-        return offline_items
-    calculation_items_future = _get_offline_recommendation(command_list, recommend_type, error_info, command_top_num)
+    # `get_recommend_from_offline_data` is an async function, so we shouldn't create a new thread for it to run in.
```

Comment:
```suggestion
    # Since the `get_recommend_from_offline_data` method is already an async function, so we don't need to create a new thread for it to run in.
```